### PR TITLE
Annotation Manager Edit and Undo Modes

### PR DESCRIPTION
**Pull request by dcupidon**
Exposed APIs that allow users to configure the annotation manager edit and undo modes.

### CODE REVIEW

**Review by sgong-pdftron — APPROVED** [review]

File: API.md [hunk 1]
```diff
@@ -1672,4 +1672,36 @@ The name of the current user. Used in the annotation manager when [`annotationMa
 config.annotationManagerEnabled = true;
 config.userId = "Bob123";
 config.userName = "Bob";
+```
+
+#### annotationManagerEditMode
+one of the [`AnnotationManagerEditMode`](./lib/constants.dart) constants, optional, default value is `AnnotationManagerEditMode.Others`
```
Comment (sgong-pdftron) [inline API.md:1678]: there isn't a "AnnotationManagerEditMode.Others" in the table below

File: API.md [hunk 2]
```diff
@@ -1672,4 +1672,36 @@ The name of the current user. Used in the annotation manager when [`annotationMa
...
+`AnnotationManagerEditMode.All` | In this mode, you can edit everyone's changes 
+
+```dart
+config.annotationManagerEnabled = true;
+config.userId = "Sam";
+config.annotationManagerEditMode = AnnotationManagerEditMode.Own;
+```
+
+#### annotationManagerUndoMode
+one of the [`AnnotationManagerUndoMode`](./lib/constants.dart) constants, optional, default value is `AnnotationManagerUndoMode.Others`
```
Comment (sgong-pdftron) [inline API.md:1694]: there isn't "others" in the table below

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Comment by dcupidon** [comment]
Fixed typo in API docs

**Review by sgong-pdftron — APPROVED** [review]

File: ios/Classes/PdftronFlutterPlugin.h [hunk 3]
```diff
@@ -351,6 +353,14 @@ static NSString * const PTViewModeColorModeKey = @"viewModeColorMode";
 static NSString * const PTInkEraserModeAllKey = @"annotationEraser";
 static NSString * const PTInkEraserModePointsKey = @"hybrideEraser";
 
+// Annotation Mananger Edit Mode
+static NSString * const PTAnnotationManagerEditModeOwnKey = @"own";
+static NSString * const PTAnnotationManagerEditModeAllKey = @"all";
+
+// Annotation Mananger Undo Mode
+static NSString * const PTAnnotationManagerUndoModeOwnKey = @"own";
+static NSString * const PTAnnotationManagerUndoModeAllKey = @"all";
```
Comment (JamesTron) [inline ios/Classes/PdftronFlutterPlugin.h:362]:
```suggestion
static NSString * const PTAnnotationManagerUndoModeOwnKey = @"undoModeOwn";
static NSString * const PTAnnotationManagerUndoModeAllKey = @"undeoModeAll";
```

File: ios/Classes/PdftronFlutterPlugin.h [hunk 4]
```diff
@@ -351,6 +353,14 @@ static NSString * const PTViewModeColorModeKey = @"viewModeColorMode";
 static NSString * const PTInkEraserModeAllKey = @"annotationEraser";
 static NSString * const PTInkEraserModePointsKey = @"hybrideEraser";
 
+// Annotation Mananger Edit Mode
+static NSString * const PTAnnotationManagerEditModeOwnKey = @"own";
+static NSString * const PTAnnotationManagerEditModeAllKey = @"all";
```
Comment (JamesTron) [inline ios/Classes/PdftronFlutterPlugin.h:358]:
```suggestion
static NSString * const PTAnnotationManagerEditModeOwnKey = @"editModeOwn";
static NSString * const PTAnnotationManagerEditModeAllKey = @"editModeAll";
```

File: ios/Classes/PTFlutterDocumentController.m [hunk 5]
```diff
@@ -1009,7 +1009,25 @@ - (void)applyViewerSettings
...
+        if ([PTAnnotationManagerEditModeOwnKey isEqualToString:self.annotationManagerEditMode]) {
+            self.toolManager.annotationManager.annotationEditMode = PTAnnotationModeEditOwn;
+        } else if ([PTAnnotationManagerEditModeAllKey isEqualToString:self.annotationManagerEditMode]) {
+            self.toolManager.annotationManager.annotationEditMode = PTAnnotationModeEditAll;
+        }
+        
+        self.toolManager.annotationAuthor = self.userId;
+        self.toolManager.annotationManager.annotationAuthorIdentifier = self.userId;
+        self.toolManager.annotationPermissionCheckEnabled = YES;
+        self.toolManager.annotationAuthorCheckEnabled = YES;
```
Comment (JamesTron) [inline ios/Classes/PTFlutterDocumentController.m:1022]: I'm not sure we should have author check enabled - @dluco ?

**Review by JamesTron — CHANGES_REQUESTED** [review]
- change strings to make the unique and somewhat more descriptive.
- check with @dluco about checking annot author.

After ^^^ is done you're good to go, no need to re-request review, just merge.

File: ios/Classes/PTFlutterDocumentController.m (hunk 5 above)
Comment (dluco) [reply ios/Classes/PTFlutterDocumentController.m:1022]: The annotation edit modes require the `annotationAuthorCheck` to be enabled in order to work - it's possible that setting the `annotationEditMode` should automatically enable the author check (in the SDK).

**Pull request merged**